tests/suites/test_suite_ssl.function

```diff
@@ -758,17 +757,26 @@ exit:
  *
  * \p endpoint_type must be set as MBEDTLS_SSL_IS_SERVER or
  * MBEDTLS_SSL_IS_CLIENT.
+ * \p pk_alg the algorithm to use, currently only MBEDTLS_PK_RSA and
+ * MBEDTLS_PK_ECDSA are supported.
+ * \p dtls_context - in case of DTLS - this is the context handling metadata.
+ * \p input_queue - used only in case of DTLS.
+ * \p output_queue - used only in case of DTLS.
  *
  * \retval  0 on success, otherwise error code.
  */
-int mbedtls_endpoint_init( mbedtls_endpoint *ep, int endpoint_type, int pk_alg )
+int mbedtls_endpoint_init( mbedtls_endpoint *ep, int endpoint_type, int pk_alg,
+                           mbedtls_test_message_socket_context *dtls_context,
+                           mbedtls_test_message_queue *input_queue,
+                           mbedtls_test_message_queue *output_queue )
 {
```

Comment:
This initialization of socket looks very complicated and I don't like it very much. When I am analyzing this code I can see that `mbedtls_test_message_socket_context` extends functionality of `mbedtls_mock_socket` but also it is his counterpart in DTLS. Maybe it would be better if we replaced `mbedtls_mock_socket socket` inside `struct mbedtls_endpoint` into `void *socket` and then assign to it dynamically `mbedtls_test_message_socket_context` in case of DTLS or `mbedtls_mock_socket` in case of TLS? Then, instead of three parameters (three NULLs in TLS), we could gave only one: endpoint must be TLS or DTLS. Did you think about it? It would require some changes and spending with it a little bit more time, but if the code would be simpler, I think it's worth. What do you think?

Reply:
It's a good idea, but it would have to go way back to first tests of the message socket. There, we'd have to add a dedicated `mbedtls_mock_msg_socket_connect` that would connect two queues, as well as call the basic `mbedtls_mock_socket_connect`. I'll be happy to add it, but given the time constraints (ping @Patater), I would suggest recording this idea as an improvement, after we introduce all the necessary changes.